### PR TITLE
Cronjob service and cronDAO fixes

### PR DESCRIPTION
**Pull request by diaswrd**
Cronjobs were not being stored in the cronDAO in the backend.

### CODE REVIEW

File: src/foam/nanos/client/Client.js [hunk 1]
```diff
@@ -276,10 +276,12 @@ foam.CLASS({
         {
           name: 'cronDAO',
           factory: function() {
-            return this.EasyDAO.create({
-              daoType: 'CLIENT',
+            return this.RequestResponseClientDAO.create({
```
Comment (kgrgreer) [inline src/foam/nanos/client/Client.js]: Why is this change needed?

File: src/foam/nanos/cron/CronScheduler.java [hunk 2]
```diff
@@ -32,13 +32,20 @@
...
    */
   private Date getMinScheduledTime() {
-    Min min = (Min) MLang.MIN(Cron.SCHEDULED_TIME);
-    cronDAO_.select(min);
-    return ((Cron) min.getValue()).getScheduledTime();
+    Min min = (Min) cronDAO_.select(
+      MLang.MIN(Cron.SCHEDULED_TIME)
+    );
+
+    if ((int) min.getValue() == 0) {
```
Comment (kgrgreer) [inline src/foam/nanos/cron/CronScheduler.java]: Here you're casting to an (int) but below you're casting to a (Date).

File: src/foam/nanos/cron/CronScheduler.java [hunk 3]
```diff
@@ -65,7 +72,10 @@ public void put(FObject obj, Detachable sub) {
         });
 
         Date minScheduledTime = getMinScheduledTime();
-        Thread.sleep(minScheduledTime.getTime() - now.getTime());
+
+        if (minScheduledTime != null) {
```
Comment (kgrgreer) [inline src/foam/nanos/cron/CronScheduler.java]: Spacing is:
 if ( minScheduledTime != null ) {

File: src/foam/nanos/cron/CronSchedulerService.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.INTERFACE({
+  package: 'foam.nanos.cron',
+  name: 'CronSchedulerService',
```
Comment (kgrgreer) [inline src/foam/nanos/cron/CronSchedulerService.js]: What is this needed for?

File: src/foam/nanos/client/Client.js (hunk 1 above)
Comment (diaswrd) [reply src/foam/nanos/client/Client.js]: I thought `RequestResponseClientDAO` was the correct DAO to connect the front-end `cronDAO` with the `cronDAO` nano in the back-end. Any suggestions?

File: src/foam/nanos/cron/CronSchedulerService.js (hunk 4 above)
Comment (diaswrd) [reply src/foam/nanos/cron/CronSchedulerService.js]: In my understanding, the `CronSchedulerService` was necessary because of its `run` method that contains a loop which verifies any cronjobs that must run at a given time. 
But I could be wrong, perhaps it doesn't need its own service? 

The original class `CronSchedulerService` author could clarify that.

File: src/foam/nanos/client/Client.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/client/Client.js]: The old code looks correct.

File: src/foam/nanos/cron/CronScheduler.java (hunk 2 above)
Comment (diaswrd) [reply src/foam/nanos/cron/CronScheduler.java]: Thanks. Pushing a fix.

File: src/foam/nanos/client/Client.js (hunk 1 above)
Comment (diaswrd) [reply src/foam/nanos/client/Client.js]: Does EasyDAO support http delegates? Like:

```
delegate: this.HTTPBox.create({
  method: 'POST',
  url: 'http://localhost:8080/cronDAO'
})
```

Accepting suggestions here.

Comment (kirkeaton) [reply src/foam/nanos/client/Client.js]: @diaswrd All you should be required to do is:
```
return this.EasyDAO.create({
  daoType: 'CLIENT',
  of: this.Cron,
  serviceName: 'cronDAO'
});
```

Was that not working?

File: src/foam/nanos/cron/CronSchedulerService.js (hunk 4 above)
Comment (kirkeaton) [reply src/foam/nanos/cron/CronSchedulerService.js]: @diaswrd Yeah I don't see why this service is necessary to be honest. CronScheduler.java implements NanoService. The run method is there because it implements Runnable.

Comment (kirkeaton) [reply src/foam/nanos/cron/CronSchedulerService.js]: @diaswrd Was this done so you could auto generate the Skeleton?

Comment (kirkeaton) [reply src/foam/nanos/cron/CronSchedulerService.js]: In any case, I would remove the run method and add
`javaExtends: [ 'java.lang.Runnable' ]` instead

File: src/foam/nanos/client/Client.js (hunk 1 above)
Comment (diaswrd) [reply src/foam/nanos/client/Client.js]: @eatonk it was working, but the changes on the GUI were not being persisted in the cronDAO service running on the server. A page reload would wipe any new cronjobs created.

**Comment by diaswrd** [comment]
@kgrgreer Client.js file removed :+1:
